### PR TITLE
fix: Go build remove unused imports

### PR DESCRIPTION
**Pull request by ocobles**
When generating the go sdk it includes some dependencies that are not used, this produces errors during the installation of the package. I added goimports in the build_go target to remove all unused imports
```
go install golang.org/x/tools/cmd/goimports@latest
cd sdk && goimports -w . && go list "$$(grep -e "^module" go.mod | cut -d ' ' -f 2)/go/..." | xargs go build
```

There are also some updates in go.mod/go.sum files after running `go work sync`, and goreleaser file updated to run clean cache

### CODE REVIEW

File: sdk/go/equinix/fabric/connection.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 	"reflect"
```
Comment (displague) [inline sdk/go/equinix/fabric/connection.go:9]: I would imagine goimports would have removed this whitespace

**Comment by displague** [comment]
Have you tried https://go.dev/ref/mod#go-mod-tidy?

**Comment by displague** [comment]
```
cd sdk && goimports -w . && go list "$$(grep -e "^module" go.mod | cut -d ' ' -f 2)/go/..." | xargs go build
```

This seems dangerous compared to `go mod tidy`. Is it being used elsewhere?

We may be picking up development/build runtime dependencies, and if those are creating a problem we could put them in a separate go.mod.

**Comment by ocobles** [comment]
I didn't see `go mod tidy` removing unused imports from code, AFAIK is just the other way around, it removes the dependency from go.mod if it is not referenced in any file

File: sdk/go/equinix/fabric/connection.go (hunk 1 above)
Comment (ocobles) [reply sdk/go/equinix/fabric/connection.go:9]: yes it was goimports

**Review by displague — APPROVED** [review]

**Pull request merged**